### DISCUSSION TITLE
matrix or lut profile

### DISCUSSION THREAD

**Discussion by SylvainArd** [ImageMagick/ImageMagick#5091] [Help]
Hello,
what type of ICC profile imagemagick use for TIFF images : lut or matrix profiles ?
Thank you
Best regards

**Comment by snibgo** [comment]
IM doesn't create ICC profiles.

IM can use profiles that are either matrix or LUT.

**Comment by SylvainArd** [comment]
OK thank you and what are the advantages or lut vs matrix ICC profiles ?
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
Any ICC profile may contain one-dimensional LUTs, one per channel, to define the Transfer Response Curve. This is useful when the TRC is not a simple mathematical function.

A 3D LUT (aka "CLUT", formally called a "multi-dimensional lookup table" is needed when a device has more than three colour channels, or where there are only three but the three primaries do not have a simple relationship to standard observer primaries. CLUTs are either very large, or sparse.

Any LUT is defined only for the range defined be the values. So they can't be used for out-of-gamut colours, and this limits the intents that can be used.

**Comment by SylvainArd** [comment]
you talked only of LUT ICC profiles, not matrix ?
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
and for measuring LAB values on scanned TIFF which we put the scanner ICC
profile, which is the best ?
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> you talked only of LUT ICC profiles, not matrix ?

Well, a matrix has the opposite properties. It is small. It is defined for all possible values. It is continuous, so no interpolation is needed. It is defined for out-of-gamut colours.

Defining an ICC profile as either "matrix" or "LUT" is too simplistic. ICC profiles can contain many elements, including matrices, 1D LUTs and 3D CLUTs. When designing a profile, we need to decide what elements are needed to achieve the required objectives, including trade-offs about performance and precision and colour fidelity and space.

> and for measuring LAB values on scanned TIFF which we put the scanner ICC profile, which is the best ?

I don't understand the question. If the scanner has a profile, then use it. If it doesn't have a profile, and you need to create one, you will need calibrated hardware to measure the source, or calibrated colour charts. Then software can create a profile for the scanner. Read the instructions from the manufacturer of the hardware, charts and software.

More information and guidance is at https://www.color.org/

**Comment by SylvainArd** [comment]
I mean what type of ICC profile for calibrating a scanner ?

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> I mean what type of ICC profile for calibrating a scanner ?

I don't understand the question.

Most scanners can create output in sRGB. Do you have a scanner? Does it create sRGB images?

**Comment by SylvainArd** [comment]
re,
so I explain what I want to do  in details  :
I want to estimate LAB values of scanned objects
for it I scan a color chart and create a profile with argyllcms
next I put this profile into a TIFF image scanned with the same scanner
and finally I measure the LAB values into this image
this is this profile I want to know that it is better matrix or LUT
thank you
best regards
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
>for it I scan a color chart and create a profile with argyllcms
next I put this profile into a TIFF image scanned with the same scanner
and finally I measure the LAB values into this image

So there are two profiles: the profile for the input device (the scanner) and a CIELab profile.

So this could be the IM command:
```
magick in.tiff +strip -profile this_profile.icc -profile Lab-D50-Identity-elle-V2.icc out.tiff
```
... where "this_profile.icc" is the profile you created.

> this is this profile I want to know that it is better matrix or LUT

Which profile? If you mean the scanner profile, follow the instructions of argyllcms. If the images you are scanning have a wide gamut, perhaps matrix is more accurate. You could create both types, and see which has the smallest error compared to the colour chart.

If you mean the CIELab profile, then that doesn't matter. That profile will be used for converting from the PCS to Lab. As the PCS is (almost certainly) either Lab or XYZ, so the conversion is either Lab to Lab (trivial) or XYZ to Lab (very easy).

**Comment by SylvainArd** [comment]
I know for the two profiles, I talk about the scanner profile, I read in
https://www.argyllcms.com/doc/Scenarios.html#PS3 that my command would that
:

" For the purposes of a poor mans colorimeter, the following would
generally be used:

colprof <https://www.argyllcms.com/doc/colprof.html> -v
<https://www.argyllcms.com/doc/colprof.html#v> -D"Scanner A"
<https://www.argyllcms.com/doc/colprof.html#E> -qm
<https://www.argyllcms.com/doc/colprof.html#q> -ax
<https://www.argyllcms.com/doc/colprof.html#a> -ua
<https://www.argyllcms.com/doc/colprof.html#u> scanner
<https://www.argyllcms.com/doc/colprof.html#p1> "

it is a  poor mans colorimeter I want to do ?

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
another person told me to do this command :
magick in.tiff +profile ICC -profile scanner_profile.icc -profile
some_lab_profile.icc -define quantum:format=floating-point -depth 32
out.tiff
without -strip, which is good ?
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> another person told me to do this command :
magick in.tiff +profile ICC -profile scanner_profile.icc -profile
some_lab_profile.icc -define quantum:format=floating-point -depth 32
out.tiff
without -strip, which is good ?

"Another person" was me, on your other thread at https://github.com/ImageMagick/ImageMagick/discussions/5054.

I haven't seen any files made by your scanner. I don't know if they need "+strip" or "+profile" or anything else.

**Comment by SylvainArd** [comment]
Ok I have not my scanner there I am, I will test this weekend
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
finally I have an example scan:
https://www.dropbox.com/s/so3vlzsyk7rgfww/feuille003.tif?dl=0
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
with this example you could find the correct command, I precise that I want
to force to delete all precedent profiles in the TIFF in the command
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
The command...
```
magick feuille003.tif -verbose info:
```
... shows your file has no embedded profile.

**Comment by SylvainArd** [comment]
yes but what I want to do is a program for measuring CieLab and my clients
could have TIFF with embedded profile, so what is the command ?

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> and my clients could have TIFF with embedded profile, so what is the command ?

We are going in circles. On the thread https://github.com/ImageMagick/ImageMagick/discussions/5054, I showed what to do if your input has an embedded profile that is correct: just use one "-profile" with a Lab ICC.

I showed what to do if there is no embedded profile: assign the correct one with "-profile", then convert to the Lab profile with a second "-profile". 

I showed what to do if it has an embedded profile but it is wrong: remove it with "+profile" or "-strip", then assign, then convert.

**Comment by SylvainArd** [comment]
ok so "+profile" or "-strip"are equivolent ?
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
other and last question : so the following command should works :
magick in.tiff +profile ICC -profile scanner_profile.icc -profile
some_lab_profile.icc -define quantum:format=floating-point -depth 32
out.tiff
?
and in this command what is the goal of : -define
quantum:format=floating-point -depth 32
and what is the meaning of ICC in : +profile ICC
thank you


Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> ok so "+profile" or "-strip"are equivolent ?

No. But either can remove any embedded ICC profiles. IM options are documented at https://imagemagick.org/script/command-line-options.php

"-define quantum:format=floating-point -depth 32" will save the output file as 32-bit floating-point values. You might, or might not, want this.

"+profile ICC" will remove ICC profiles, and not other profiles.

**Comment by SylvainArd** [comment]
OK thank you
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
so if I want to get LAB real values of TIFF generated by "-define
quantum:format=floating-point -depth 32
I need to do what to get :
L from 0 to 100
A and B from -128 to 127
floating points values are signed or not ?

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
I have answered that in two of your other posts.

If you have new questions, please ask them. But don't keep repeating questions I have already answered.

**Comment by SylvainArd** [comment]
yes but you talked about QuantumRange but it is for integers I suppose not
floating points
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
"-define quantum:format=floating-point -depth 32" will save the output file as 32-bit floating-point values. That has no impact on how pixel values are stored in memory. In memory, the nominal values are zero to QuantumRange, or -QuantumRange/2 to +QuantumRange/2 in the case of a* and b* when using a Lab profile.

**Comment by SylvainArd** [comment]
OK thank you, I will test this weekend
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers
